### PR TITLE
Prettier

### PR DESCRIPTION
**Pull request by Hamms**
Adds Prettier to our codebase!

- enforced [with eslint](https://prettier.io/docs/en/eslint.html)
- configured to most closely match our existing styles
- instructions for editor support added to Setup.md

Note that we expect this PR to fail circle tests, since it adds rules but doesn't update code to match them. If y'all sign off on this implementation, I'll open a second PR that performs the actual code reformatting.

### CODE REVIEW

**Review by islemaster — APPROVED** [review]
I fully endorse us moving this direction. It sounds like people can set up their local configuration to do this on hotkey or on save per personal preference.  I haven't actually tried this with my editor yet but looking forward to it - I'll let you know when I do.

File: SETUP.md [hunk 1]
```diff
@@ -187,6 +187,20 @@ If waiting around for javascript builds is making you sad, consider sending buil
...
 
+## Editor configuration
+
+We enforce linting rules for all our code, and we recommend you set up your editor to integrate with that linting.
+
+### Javascript
+
+We use [eslint](https://eslint.org/) to lint our Javascript; see [the official integrations guide](https://eslint.org/docs/user-guide/integrations) for instructions for your editor of choice.
+
+Our lint configuration uses formatting rules provided by [Prettier](https://prettier.io/). You can configure your editor to auto-format your code to meet our requirements, in addition to the error highlighting provided by eslint. See [the official integrations guide](https://prettier.io/docs/en/editors.html) for instructions for your editor of choice.
```
Comment (davidsbailey) [inline SETUP.md:198]: with this change, how does linting work at a conceptual level -- is prettier simply providing a set of lint rules to eslint, and letting eslint do the actual linting?

**Review by davidsbailey — APPROVED** [review]
I am very excited for this change! as you're probably aware, https://github.com/code-dot-org/code-dot-org/pull/26885 touches a lot of files for the react 16 upgrade, so you will probably want to merge staging into your branch after that PR goes in before running prettier on all our JS files.

File: SETUP.md (hunk 1 above)
Comment (Hamms) [reply SETUP.md:198]: Yep!

**Pull request merged**